src/Transit/Internal/Network.hs

```diff
@@ -111,32 +114,25 @@ getSocketPort (Just sock') = do
   portnum <- socketPort sock'
   return (Just portnum)
 
-type Hostname = Text
-
-ipv4ToHostname :: Word32 -> Hostname
-ipv4ToHostname ip =
-  let (q1, r1) = ip `divMod` 256
-      (q2, r2) = q1 `divMod` 256
-      (q3, r3) = q2 `divMod` 256
-  in
-    show r1 <> "." <> show r2 <> "." <> show r3 <> "." <> show q3
-
 buildDirectHints :: PortNumber -> IO (Set.Set ConnectionHint)
 buildDirectHints portnum = do
   nwInterfaces <- getNetworkInterfaces
   let nonLoopbackInterfaces =
-        filter (\nwInterface ->
-                   let (IPv4 addr4) = ipv4 nwInterface
-                   in
-                     (ipv4ToHostname addr4 /= "0.0.0.0")
-                     && (ipv4ToHostname addr4 /= "127.0.0.1"))
-        nwInterfaces
-  return $ Set.fromList $ map (\nwInterface ->
-                                 let (IPv4 addr4) = ipv4 nwInterface in
-                                   Direct Hint { hostname = ipv4ToHostname addr4
-                                               , port = fromIntegral portnum
-                                               , priority = 0
-                                               , ctype = DirectTcpV1 }) nonLoopbackInterfaces
+          filter (\nwInterface -> name nwInterface /= "lo" &&
+                                  (show (ipv4 nwInterface) /= ("0.0.0.0" :: Text) ||
+                                   show (ipv6 nwInterface) /= ("0:0:0:0:0:0:0:0" :: Text)))
```

Comment:
https://hackage.haskell.org/package/gi-gio-2.0.30/docs/GI-Gio-Objects-InetAddress.html#v:inetAddressGetIsAny would probably be a good way to check this ... except that it's from gi-gio.  Also the implementation looks like a cheater (just compares against 0 instead of INADDR_ANY, IN6ADDR_ANY, etc).  Though it will probably always get away with it.

You could probably compare against the Word32s inside IPv4 and IPv6 here ... but ... that's probably not really a win over this version.  Oh well.  Nothing to do here I guess.